src/ray/raylet/agent_manager.h

```diff
@@ -70,7 +93,7 @@ class DefaultAgentManagerServiceHandler : public rpc::AgentManagerServiceHandler
   }
 
  private:
-  std::unique_ptr<AgentManager> &delegate_;
+  std::shared_ptr<AgentManager> &delegate_;
```

Comment:
Why shared pointer here? Btw, before this change, it's a reference unique ptr which is also not good. Either a reference or a raw ptr should be good here.

Reply:
Maybe I can redefine to `std::shared_ptr<AgentManager> delegate_`? I use shared_ptr because I should access AgentManager in worker_pool.

Reply:
I know the reason why a ptr reference before. It is because  the construction of `DefaultAgentManagerServiceHandler` is happened before `AgentManger` construction.